hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/RandomKeyGenerator.java

```diff
@@ -622,7 +642,11 @@ public void run() {
                 try (Scope writeScope = GlobalTracer.get()
                     .buildSpan("writeKeyData")
                     .startActive(true)) {
-                  os.write(keyValue);
+                  for (long nrRemaining = keySize - randomValue.length;
+                        nrRemaining > 0; nrRemaining -= bufferSize) {
+                    int curSize = (int)Math.min(bufferSize, nrRemaining);
+                    os.write(keyValueBuffer, 0, curSize);
```

Comment:
Can we generate random string here and update the digest inline up to the buffersize? 
This way, we will not be writing the same buffer repeatedly that will be cached in OS buffer/cache.

Reply:
Thanks， I'll correct it,  after that every key will have an unique UUID.

Reply:
You are right. No issue at the socket layer. I'm thinking of the DN side, the chunk files of the same key being written could be the same in this scheme. That might increase the write performance compared with 2GB fully random chunks.  As long we use it consistently, it should be fine. Later on, we will can an option to write 0 only by default and  random up to buffersize when a parameter is specified.